datasources/whatsapp/whatsapp.go

```diff
@@ -0,0 +1,183 @@
+// Imports from individual WhatsApp conversation exports: https://faq.whatsapp.com/1180414079177245/
+package whatsapp
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/timelinize/timelinize/timeline"
+	"go.uber.org/zap"
+)
+
+func init() {
+	err := timeline.RegisterDataSource(timeline.DataSource{
+		Name:            "whatsapp",
+		Title:           "WhatsApp",
+		Icon:            "whatsapp.svg",
+		Description:     "A WhatsApp Chat Export containing (most of the) information from a chat between a fixed set of people",
+		NewFileImporter: func() timeline.FileImporter { return new(Importer) },
+	})
+	if err != nil {
+		timeline.Log.Fatal("registering data source", zap.Error(err))
+	}
+}
+
+// Importer can import the data from a zip file or folder.
+type Importer struct{}
+
+// Recognize returns whether the input is supported.
+func (Importer) Recognize(_ context.Context, dirEntry timeline.DirEntry, _ timeline.RecognizeParams) (timeline.Recognition, error) {
+	if dirEntry.FileExists(chatPath) {
+		return timeline.Recognition{Confidence: 1}, nil
+	}
+	return timeline.Recognition{}, nil
+}
+
+// FileImport imports data from the file or folder.
+func (i *Importer) FileImport(_ context.Context, dirEntry timeline.DirEntry, params timeline.ImportParams) error {
+	file, err := dirEntry.FS.Open(chatPath)
+	if err != nil {
+		return fmt.Errorf("loading chat: %w", err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(chatSplit)
+
+	// We have to store all and emit later, because we have no way of knowing
+	// who's in the conversation until we've parsed it
+	var messages []*timeline.Graph
+	recipients := make(map[string]timeline.Entity)
+
+	for scanner.Scan() {
+		messageLine := scanner.Text()
+
+		sub := messageStartRegex.FindStringSubmatch(messageLine)
+
+		timestamp, err := parseTime(sub[2], sub[3])
+		if err != nil {
+			return err
+		}
+
+		content := strings.Split(messageLine[len(sub[0])-1:], "\u200E")
+		name := sub[4]
+
+		owner := timeline.Entity{
+			Name: name,
+			Attributes: []timeline.Attribute{
+				{
+					Name:     "WhatsApp Name",
```

Comment:
```suggestion
					Name:     "whatsapp_name",
```

I should clarify in the docs that attribute names are snake_cased... they're mostly internal and act kind of like keys in the DB; though the UI can have a mapping to human-friendly names for common attributes.

Reply:
Ahh! I started with the snake_case, but saw it come through as snake in the UI, so swapped it 😅 Thanks!